vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/vaadin-grid-flow-selection-column.js

```diff
@@ -1,156 +1,97 @@
-     *
-     * @override
-     */
-    _defaultHeaderRenderer(root, _column) {
-      let checkbox = root.firstElementChild;
-      if (!checkbox) {
-        checkbox = document.createElement('vaadin-checkbox');
-        checkbox.id = 'selectAllCheckbox';
-        checkbox.setAttribute('aria-label', 'Select All');
-        checkbox.classList.add('vaadin-grid-select-all-checkbox');
-        checkbox.addEventListener('click', this._onSelectAllClick.bind(this));
-        root.appendChild(checkbox);
-      }
+class GridFlowSelectionColumn extends GridSelectionColumnBaseMixin(GridColumn) {
 
-      const checked = this.selectAll;
-      checkbox.hidden = this.selectAllHidden;
-      checkbox.checked = checked;
-      checkbox.indeterminate = this.indeterminate;
-    }
+  static get is() {
+    return 'vaadin-grid-flow-selection-column';
+  }
 
-    /**
-     * Renders the Select Row checkbox to the body cell.
-     *
-     * @override
-     */
-    _defaultRenderer(root, _column, { item, selected }) {
-      let checkbox = root.firstElementChild;
-      if (!checkbox) {
-        checkbox = document.createElement('vaadin-checkbox');
-        checkbox.setAttribute('aria-label', 'Select Row');
-        checkbox.addEventListener('click', this._onSelectClick.bind(this));
-        root.appendChild(checkbox);
+  static get properties() {
+    return {
+      /**
+       * Override property to enable auto-width
+       */
+      autoWidth: {
+        type: Boolean,
+        value: true
+      },
+
+      /**
+       * Override property to set custom width
+       */
+      width: {
+        type: String,
+        value: '56px'
       }
+    };
+  }
 
-      checkbox.__item = item;
-      checkbox.checked = selected;
-    }
-
-    _onSelectClick(e) {
-      e.currentTarget.checked ? this._grid.$connector.doDeselection([e.currentTarget.__item], true) : this._grid.$connector.doSelection([e.currentTarget.__item], true);
+  /**
+   * Override method from `GridSelectionColumnBaseMixin` to add ID to select all
+   * checkbox
+   *
+   * @override
+   */
+  _defaultHeaderRenderer(root, _column) {
+    super._defaultHeaderRenderer(root, _column);
+    const checkbox = root.firstElementChild;
+    if (checkbox) {
+      checkbox.id = 'selectAllCheckbox';
     }
+  }
 
-    _onSelectAllClick(e) {
-      e.preventDefault();
-      if (this._grid.hasAttribute('disabled')) {
-        e.currentTarget.checked = !e.currentTarget.checked;
-        return;
-      }
-      this.selectAll ? this.$server.deselectAll() : this.$server.selectAll();
-    }
 
-    _onSelectEvent(e) {
-    }
+  /**
+   * Override a method from `GridSelectionColumnBaseMixin` to handle the user
+   * selecting all items.
+   *
+   * @protected
+   * @override
+   */
+  _selectAll() {
+    this.selectAll = true;
+    this.$server.selectAll();
+  }
 
-    _onDeselectEvent(e) {
-      if (e.detail.userOriginated) {
-        this.selectAll = false;
-      }
-    }
```

Comment:
Did this work at all? I can't find neither `select` nor `deselect` events in the web component.

Reply:
As of this PR, we preemptively update `this.selectAll` on the client-side once the user clicks Select All / Deselect All. I wonder if this change can affect anything.

Reply:
No those didn't work. Maybe it was part of the connector at some point 🤔.

Anyway, all it would have done would be to optimistically deselect the select all checkbox, so I adopted that into `deselectItem`. Doesn't seem to cause any harm with the existing tests.